crates/opencascade/src/primitives.rs

```diff
@@ -298,6 +297,33 @@ impl Wire {
     pub fn freeform() {}
 }
 
+pub struct Surface {
+    inner: UniquePtr<ffi::HandleGeomSurface>,
+}
+
+impl Surface {
+    pub fn bezier(poles: &[&[DVec3]]) -> Self {
```

Comment:
We could make this API a bit more flexible:

```suggestion
    pub fn bezier<T: AsRef<[DVec3]>>(poles: impl IntoIterator<Item = T>) -> Self {
```

You'll then need to update various points in this function with calls to `.into_iter()` and `.as_ref()`

This should allow more flexibility in calling this function, allowing things like:

```rust
let surface = Surface::bezier([points1, points2]);
```

Reply:
Or maybe we can have a doubly nested `impl Iterator`? I'll look into that.

Reply:
@julianschuler ah, you can collect the iterator into a `Vec` for ease-of-use internally. It's more about presenting a nice API to end users and less about performance.

Reply:
Thanks for the clarification, I now implemented it that way.